src/Uno.UI.Composition/Composition/CompositionSpriteShape.skia.cs

```diff
@@ -126,9 +136,38 @@ private static SKPaint TryCreateAndClearPaint(ref SKPaint? paint, bool isStroke,
 					pathEffect.Dispose();
 					paint.PathEffect = null;
 				}
+
+				if (paint.ImageFilter is { } imageFilter)
+				{
+					imageFilter.Dispose();
+					paint.ImageFilter = null;
+				}
+
+				if (paint.MaskFilter is { } maskFilter)
+				{
+					maskFilter.Dispose();
+					paint.MaskFilter = null;
+				}
+
+				if (paint.Typeface is { } typeface)
+				{
+					typeface.Dispose();
+					paint.Typeface = null;
+				}
 			}
 
+			paint.IsStroke = isStroke;
+
+			// uno-specific defaults
+			paint.Color = SKColors.White;   // Transparent color wouldn't draw anything
+			paint.IsAutohinted = true;
+			// paint.IsAntialias = true; // IMPORTANT: don't set this to true by default. It breaks canvas clipping on Linux for some reason.
```

Comment:
This is super super curious. Antialiasing it generally an important thing for us, I think.

Reply:
It is indeed :/. The point of this change is to set antialiasing as needed instead of enabling it by default, which still sounds silly but it seems to be the only somewhat reasonable fix. Removing this method entirely and creating a new SKPath when needed would also fix this. I might try to remove it and profile the difference.

Reply:
Does it still happen with SkiaSharp 3 ?

Reply:
I didn't think to try that at all. It works fine with SkiaSharp 3!

Reply:
The `TryCreateAndClear***` methods that looked hacky to me above. Unfortunately, profiling should a significant impact when I removed them.

While investigating the problem, I found that we can actually get away with not allocating almost any `SKPaint`s across the entire app. Almost all our `SKPaint` usages are very temporary, usually just to immediately draw something. So, we can "borrow" `SKPaint` instances when needed and it turns out that there's an `SKPaint.Reset` method that cleans the `SKPaint` instances between usages, so this also takes care of these `TryCreateAndClear***` methods that can now soundly depend on `Reset` instead of the error-prone manual cleanup that we did before.

At this point, this is mostly a perf PR, that just so happens to also fix the problem with clipping on Linux.

Reply:
Update: even after all the changes, the problem still reproduces on Linux, unless we disable anti-aliasing, which is a non-starter as Youssef said, or unless we close the SKPath everytime we use it for painting (so basically the same as remove the `TryCreate***` methods, which turned out to be very impactful).

For now, the only real fix is to move to SkiaSharp 3